src/py/reactpy/reactpy/_option.py

```diff
@@ -8,6 +8,13 @@
 
 _O = TypeVar("_O")
 logger = getLogger(__name__)
+UNDEFINED = cast(Any, object())
+
+
+def deprecate(option: Option[_O], message: str) -> Option[_O]:
+    option.__class__ = _DeprecatedOption
```

Comment:
Problem: Directly assigning __class__ is not recommended. Use inheritance instead.